crates/terminal_view/src/terminal_panel.rs

```diff
@@ -955,6 +961,42 @@ impl TerminalPanel {
             is_enabled_in_workspace(workspace.read(cx), cx)
         })
     }
+
+    fn set_position_right(
+        workspace: &mut Workspace,
+        _action: &DockRight,
+        cx: &mut ViewContext<Workspace>,
+    ) {
+        let Some(terminal_panel) = workspace.panel::<Self>(cx) else {
+            return;
+        };
+
+        terminal_panel.set_position(DockPosition::Right, cx);
+    }
+
+    fn set_position_bottom(
+        workspace: &mut Workspace,
+        _action: &DockBottom,
+        cx: &mut ViewContext<Workspace>,
+    ) {
+        let Some(terminal_panel) = workspace.panel::<Self>(cx) else {
+            return;
+        };
+
+        terminal_panel.set_position(DockPosition::Bottom, cx);
+    }
+
+    fn set_position_left(
+        workspace: &mut Workspace,
+        _action: &DockLeft,
+        cx: &mut ViewContext<Workspace>,
+    ) {
+        let Some(terminal_panel) = workspace.panel::<Self>(cx) else {
+            return;
+        };
+
+        terminal_panel.set_position(DockPosition::Left, cx);
+    }
```

Comment:
This is a bit repetitive but, since I'm not good at Rust, I didn't manage to find a way to have a single function that would be triggered for any of the Dock* actions and that would then match on the action and pick the correct DockPosition enum value. If you're aware of a better way to do this, please share it ✌️